src/react/hooks/__tests__/useBackgroundQuery.test.tsx

```diff
@@ -504,10 +502,9 @@ function renderPaginatedIntegrationTest({
         <ErrorBoundary
           fallback={<div>Error</div>}
           onError={(error) => {
-            ProfiledApp.updateSnapshot((snapshot) => ({
-              ...snapshot,
-              errorCount: snapshot.errorCount + 1,
-              errors: snapshot.errors.concat(error),
+            ProfiledApp.setSnapshot(({ errorCount, errors }) => ({
+              errorCount: errorCount + 1,
+              errors: errors.concat(error),
```

Comment:
I'm almost considering that we get `immer` in here for the tests.

That would end up looking like this,

```suggestion
            ProfiledApp.updateSnapshot(snapshot => {
              snapshot.errorCount++;
              snapshot.errors.push(error)
            
```

but would still enable something like this
```js
            ProfiledApp.updateSnapshot(snapshot => {
              return {}
            })
```
which would reset the snapshot to an empty object - which would not be possible with the `setSnapshot` notation here.

Reply:
I guess I'm indifferent here. I didn't consider an alternative library only because it was a small change, but I can see how it would be useful.

FWIW, to reset back to an empty object, I figured you'd just use `updateSnapshot` directly since `updateSnapshot` does a whole replacement on the snapshot object.

```ts
ProfiledApp.updateSnapshot({})
```

I never intended `setSnapshot` to be used for this case, and IMO was the differentiator between the `update*` and `set*` versions of this function.

The reason I felt like a `setSnapshot` would be useful (which you don't see in this PR since this PR contains updates to coutns, but would be used in `useLoadableQuery`) is cases where you want to record the value from a hook, which might only be part of the whole snapshot.

```ts
function Child({ queryRef }) {
  const { data } = useReadQuery(queryRef)

  // No need for the function callback since I can just update the partial snapshot
  ProfiledApp.setSnapshot({ data })
}

// or record the whole object under a specific key without the need for a callback function
ProfiledApp.setSnapshot({ result: useReadQuery(queryRef) })
```

Again, I'm indifferent, and if adding a dependency on `immer` would make this more useful, thats fine.

Thoughts?

Reply:
Actually, I'm just now realizing that `produce` in Immer enforces a callback function. 

The motivation behind `setSnapshot` was to try and avoid the noise of a callback function when I just wanted to record a new value for a particular key. This PR doesn't do a great job of highlighting this use case since `updateSnapshot` is mainly used to increment counters, but there are a decent number of cases in the `useLoadableQuery` PR that would benefit from the `setSnapshot` style.

```ts
const result = useReadQuery(queryRef)

ProfiledApp.setSnapshot({ result })

// vs
ProfiledApp.updateSnapshot((snapshot) => ({ ...snapshot, result }))

// or with immer
ProfiledApp.updateSnapshot((snapshot) => {
  snapshot.result = result;
})
```

This is especially useful if you just want to inline everything:

```ts
ProfiledApp.setSnapshot({ result: useReadQuery(queryRef) })
```

I'd like to preserve _some_ way of being able to partial update a key on the snapshot without the use of a callback function if possible to cut down on the noise. I do agree that the immer-style way of doing things is useful for incrementing counters or "mutating" existing values. Perhaps there is a way we can have both?